frog/imports/ui/Home.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { Component, PropTypes } from 'react';
+import ReactDOM from 'react-dom';
+import { Meteor } from 'meteor/meteor';
+import { createContainer } from 'meteor/react-meteor-data';
+import { Messages } from '../api/messages.js';
+
+export default class Home extends Component {
+
+  renderUsers(){
+    return (
+      this.props.presentUsers ? 
+      this.props.presentUsers
+        .map((user)=> user.userId)
+        .map((userId) =>
+          <li key={userId}>{Meteor.users.findOne({_id: userId}).username}</li>
+      ) : <li>empty</li>
+    );
+  }
+
+  handleMessage(event) {
+    event.preventDefault();
+  }
+
+  handleMessageSubmit(event) {
+    event.preventDefault();
+    const text = ReactDOM.findDOMNode(this.refs.msg).value.trim();
+    ReactDOM.findDOMNode(this.refs.msg).value = "";
+    if (text!= ""){
+      Meteor.call('messages.insert', text);
+    }
+  }
+
+  renderMessages(){
+    return (
+      this.props.messages ?
+      this.props.messages.map((msg) => (
+        <li key={msg.createdAt}><b>{msg.username} on {msg.createdAt.toUTCString()}:</b> {msg.text}</li>))
```

Comment:
Will there be a key collision issue if two messages are created at the same second ? Maybe a safer key would be the concatenation of `username` and `createdAt`.